snapcraft/cli/_runner.py

```diff
@@ -62,18 +95,8 @@
 def run(ctx, debug, catch_exceptions=False, **kwargs):
     """Snapcraft is a delightful packaging tool."""
 
-    # Do this early, if we are in managed-host and not root yet we want to re-exec with
-    # sudo keeping the current environment settings.
-    if os.getenv("SNAPCRAFT_BUILD_ENVIRONMENT") == "managed-host" and os.geteuid() != 0:
-        snap_path = os.getenv("SNAP")
-        # Use -E to keep the environment.
-        cmd = ["usr/bin/sudo", "-E"]
-        # Pick the correct interpreter if running from a snap (in managed-host mode, this
-        # is most likely the only scenario).
-        if os.getenv("SNAP_NAME") == "snapcraft":
-            cmd.append(os.path.join(snap_path, "usr", "bin", "python3"))
-        cmd.extend(sys.argv)
-        os.execve("/usr/bin/sudo", cmd, os.environ)
+    if _is_legacy_reexec():
+        _legacy_reexec()
```

Comment:
Given that this calls os.execv, mind adding a comment here that this will not return so future readers don't get confused?